src/core/ext/xds/xds_client.cc

```diff
@@ -1316,8 +1319,15 @@ void XdsClient::ChannelState::AdsCallState::OnStatusReceivedLocked(
     // Try to restart the call.
     parent_->OnCallFinishedLocked();
     // Send error to all watchers.
-    xds_client()->NotifyOnErrorLocked(
-        GRPC_ERROR_CREATE_FROM_STATIC_STRING("xds call failed"));
+    char* status_details = grpc_slice_to_c_string(status_details_);
```

Comment:
Instead, please use `StringViewFromSlice()`, to avoid an unnecessary allocation.

https://github.com/grpc/grpc/blob/fe913387da494713a945f7842669bd3bf3c27a03/src/core/lib/slice/slice_internal.h#L92

Note that `absl::StrFormat()` knows how to handle `absl::string_view` parameters, so you can probably just inline this into that call instead of storing it in a separate variable.

Reply:
done